src/idfxt/JSONDataInterface.cpp

```diff
@@ -0,0 +1,209 @@
+{
+    if (attribute) {
+        string attribute_name = attribute->string;
+        if (attribute_name.find("imum")  != std::string::npos) { //only if range checkable fields
+            double attribute_value =  (attribute->valuestring) ? stod(attribute->valuestring) : attribute->valueint;
+            if (attribute_name == "exclude_minimum") {
+                if (!(property_value > attribute_value)) {
+                    valid = false;
+                    cout << "INVALID: " << child_name << " : " << property_name << " : " << property_value <<  " must be > " << attribute_value << endl;
+                }
+            } else if (attribute_name == "exclude_maximum") {
+                if (!(property_value < attribute_value)) {
+                    valid = false;
+                    cout << "INVALID: " << child_name << " : " << property_name << " : " << property_value <<  " must be < " << attribute_value << endl;
+                }
+            } else if (attribute_name == "minimum") {
+                if (!(property_value >= attribute_value)) {
+                    valid = false;
+                    cout << "INVALID: " << child_name << " : " << property_name << " : " << property_value <<  " must be >= " << attribute_value << endl;;
+                }
+            } else if (attribute_name == "maximum") {
+                if (!(property_value <= attribute_value)) {
+                    valid = false;
+                    cout << "INVALID: " << child_name << " : " << property_name << " : " << property_value <<  " must be <= " << attribute_value << endl;
+                }
+            }
+        }
+    }
+}
+
+void JSONDataInterface::checkNumeric(double property_value, string property_name, cJSON *schema_object, bool &valid, string child_name)
+{
+    if (schema_object) {                               //get matching schema property
+        cJSON *schema_property = cJSON_GetObjectItem(schema_object, property_name.c_str());
+        if (schema_property) {
+            for (int a = 0; a < cJSON_GetArraySize(schema_property); a++) {
+                cJSON *attribute = cJSON_GetArrayItem(schema_property, a);
+                checkRange(attribute, property_name, child_name, valid, property_value);
+            }
+        }
+    }
+}
+
+bool JSONDataInterface::validateModel()
+{
+    bool valid = true;
+    cJSON *child = model_j->child;
+    if (child) {
+        do {
+            string child_name = child->string;
+            for (int i = 0; i < cJSON_GetArraySize(child); i++) {
+                cJSON *property = cJSON_GetArrayItem(child,i);
+                if (property) {
+                    string property_name = property->string;
+                    switch (property->type) {
+                    case cJSON_Number: { //check numeric types
+                        double property_value = (property->valuedouble) ? property->valuedouble : property->valueint;
```

Comment:
I just meant that you are converting all ints to doubles and then validating that way.  It would be better to validate ints as ints